docs/manual/resource-leak-checker.tex

```diff
@@ -0,0 +1,343 @@
+Non-constructor methods may also create obligations
+when re-assigning non-final owning fields or allocating
+new system-level resources.
+
+A post-condition annotation,
+\<@CreatesMustCallFor>,
+indicates for which expression an obligation is created.
+If you write \<@CreatesMustCallFor(>\emph{T}\<)> on a method \emph{M} that
+overrides a method \emph{N}, then \emph{N} must also be annotated as
+\<@CreatesMustCallFor(>\emph{T}\<)>.  (\emph{N} may also have other
+\<@CreatesMustCallFor> annotations that \emph{M} does not.)
+If you write \<@CreatesMustCallFor> on a method that does not actually
+create an obligation, the checker may issue false positive warnings at calls to
+that method, but it never causes
+missed warnings (i.e., the checker is sound).
+
+\<@CreatesMustCallFor> allows the Resource Leak Checker to verify uses of non-final fields
+that contain a resource, even if they are re-assigned. Consider
+the following example:
+
+\begin{verbatim}
+  @MustCall("close") // default qualifier for uses of SocketContainer
+  class SocketContainer {
+    private @Owning Socket sock;
+
+    public SocketContainer() { sock = ...; }
+
+    void close() { sock.close() };
+
+    @CreatesMustCallFor("this")
+    void reconnect() {
+      if (!sock.isClosed()) {
+        sock.close();
+      }
+      sock = ...;
+    }
+  }
+\end{verbatim}
+
+In the lifetime of a \<SocketContainer> object, \<sock>
+might be re-assigned arbitrarily many times: once at each
+call to \<reconnect()>. This code is safe, however: \<reconnect()>
+ensures that \<sock> is closed before re-assigning it.
+
+Support for \<@CreatesMustCallFor> can be disabled by passing the \<-AnoCreatesMustCallFor> command-line
+argument to the checker. In this mode, \<@CreatesMustCallFor> annotations on methods are ignored at call-sites,
+and re-assigning a non-final, owning field is always an error.
+
+Sections~\ref{resource-leak-createsmustcallfor-callsite}
+and~\ref{resource-leak-createsmustcallfor-declaration}
+explain how the Resource Leak Checker verifies uses and declarations of
+methods annotated with \<@CreatesMustCallFor>.
+
+
+\subsectionAndLabel{Requirements at a call site of a \<@CreatesMustCallFor> method}{resource-leak-createsmustcallfor-callsite}
+
+At a call site to a method annotated as
+\<@CreatesMustCallFor(>\emph{expr}\<)>, the Resource Leak Checker:
+\begin{enumerate}
+\item
+  Treats any existing \<@MustCall> obligations of \emph{expr} as \emph{satisfied},
+\item
+  Creates a fresh obligation to check, as if \emph{expr} was a newly-allocated local
+  variable.
+\item
+  Requires that the expression corresponding to \emph{expr} (that is, \emph{expr}
+  viewpoint-adapted to the method call site) is owned; that is, it is
+  annotated or defaulted as \<@Owning>.  Otherwise, the checker
+  will issue a \<reset.not.owning> error at the call-site. You can avoid this
+  error by extracting the relevant expression into a new local variable (because
```

Comment:
The new local variable is near the call site.
Then what?  I think the new local variable should be passed as an actual argument.  But in that case the local variable might be just a prefix of _expr_ rather than the whole thing.